src/riak_core_util.erl

```diff
@@ -704,6 +712,124 @@ proxy(Parent, Fun) ->
+enable_job_class(Class) ->
+    case app_helper:get_env(riak_core, job_accept_class) of
+        [_|_] = EnabledClasses ->
+            case lists:member(Class, EnabledClasses) of
+                true ->
+                    ok;
+                _ ->
+                    application:set_env(
+                        riak_core, job_accept_class, [Class | EnabledClasses])
+            end;
+        _ ->
+            application:set_env(riak_core, job_accept_class, [Class])
+    end.
+
+-spec disable_job_class(term()) -> ok | {error, term()}.
+%% @doc Internal API to disable the specified job class.
+%% WARNING:
+%% * This function may not remain in this form once the Jobs API is live!
+%% * Parameter types ARE NOT validated by the same rules as the public API!
+%% You are STRONGLY advised to use disable_job_class/2.
+disable_job_class(Class) ->
+    case app_helper:get_env(riak_core, job_accept_class) of
+        [_|_] = EnabledClasses ->
+            case lists:member(Class, EnabledClasses) of
+                false ->
+                    ok;
+                _ ->
+                    application:set_env(riak_core, job_accept_class,
+                        lists:delete(Class, EnabledClasses))
+            end;
+        _ ->
+            ok
+    end.
+
+-spec job_class_enabled(term()) -> boolean().
+%% @doc Internal API to determine whether to accept/reject a job.
+%% WARNING:
+%% * This function may not remain in this form once the Jobs API is live!
+%% * Parameter types ARE NOT validated by the same rules as the public API!
+%% You are STRONGLY advised to use job_class_enabled/2.
+job_class_enabled(Class) ->
+    case app_helper:get_env(riak_core, job_accept_class) of
+        undefined ->
+            true;
+        [] ->
+            false;
+        [_|_] = EnabledClasses ->
+            lists:member(Class, EnabledClasses);
+        Other ->
+            % Don't crash if it's not a list - that should never be the case,
+            % but since the value *can* be manipulated externally be more
+            % accommodating. If someone mucks it up, nothing's going to be
+            % allowed, but give them a chance to catch on instead of crashing.
+            _ = lager:error(
+                "riak_core.job_accept_class is not a list: ~p", [Other]),
+            false
+    end.
+
+-spec job_class_disabled_message(atom(), term()) -> binary() | string().
+%% @doc The error message to be returned to a client for a disabled job class.
+%% WARNING:
+%% * This function is likely to be extended to accept a Job as well as a Class
+%%   when the Jobs API is live.
+job_class_disabled_message(binary, Class) ->
+    erlang:list_to_binary(job_class_disabled_message(text, Class));
+job_class_disabled_message(text, Class) ->
+    lists:flatten(io_lib:format("Operation '~p' is not enabled", [Class])).
+
+-spec report_job_request_disposition(
+    boolean(), term(), module(), atom(), pos_integer(), term())
```

Comment:
The meaning of parameters (especially the true/false in the function heads below) would be much easier to figure out if we used parameter names in this spec, rather than just types... maybe something like
```erlang
-spec report_job_request_disposition(
    JobAccepted :: boolean(), 
    JobClass :: term(), 
    CallingModule :: module(), 
    CallingFunction :: atom(), 
    CallingLine :: pos_integer(), 
    Client :: term()), %% Still not exactly sure what "Client" is - ClientIdentifier? "HTTP/PB"?
) -> ok | {error, term()}.
```